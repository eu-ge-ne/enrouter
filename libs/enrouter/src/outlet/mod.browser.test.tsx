import type { FC, PropsWithChildren } from "react";
import { describe, test, expect } from "vitest";
import { render } from "vitest-browser-react";
import * as regexparam from "regexparam";

import { renderMatches } from "#render/mod.js";
import { Outlet } from "./mod.js";

import type { RouteMatch } from "#match/mod.js";

const wrapperId = "test-wrapper";

const wrapper: FC<PropsWithChildren> = ({ children }) => (
  <div data-testid={wrapperId}>{children}</div>
);

describe("outlets", () => {
  test("using index elements", async () => {
    const matches: RouteMatch[] = [
      {
        handler: {
          route: {
            path: "/",
            mod: [],
            link: [[], []],
          },
          test: regexparam.parse("/", true),
          modules: [],
          layout: {
            main: (
              <div>
                <div>Layout</div>
                <Outlet name="main" />
              </div>
            ),
          },
          index: {
            main: <div>Index</div>,
          },
        },
        location: "/",
        isFull: true,
        params: {},
      },
    ];

    const screen = render(renderMatches(matches), { wrapper });

    await expect.element(screen.getByTestId(wrapperId)).toBeVisible();
    expect(screen.container).toMatchSnapshot();
  });

  test("using next layout elements", async () => {
    const matches: RouteMatch[] = [
      {
        handler: {
          route: {
            path: "/",
            mod: [],
            link: [[], []],
          },
          test: regexparam.parse("/", true),
          modules: [],
          layout: {
            main: (
              <div>
                <div>Layout</div>
                <Outlet name="main" />
              </div>
            ),
          },
        },
        location: "/",
        isFull: false,
        params: {},
      },
      {
        handler: {
          route: {
            path: "/a",
            mod: [],
            link: [[], []],
          },
          test: regexparam.parse("/a", true),
          modules: [],
          layout: {
            main: (
              <div>
                <div>Next layout</div>
              </div>
            ),
          },
        },
        location: "/a",
        isFull: true,
        params: {},
      },
    ];
    matches[0]!.next = matches[1];

    const screen = render(renderMatches(matches), { wrapper });

    await expect.element(screen.getByTestId(wrapperId)).toBeVisible();
    expect(screen.container).toMatchSnapshot();
  });
});
