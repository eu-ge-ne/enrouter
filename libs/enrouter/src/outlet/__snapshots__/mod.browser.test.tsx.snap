// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`outlets > using index elements 1`] = `
<div>
  <div
    data-testid="test-wrapper"
  >
    <div>
      <div>
        Layout
      </div>
      <div>
        Index
      </div>
    </div>
  </div>
</div>
`;

exports[`outlets > using next layout elements 1`] = `
<div>
  <div
    data-testid="test-wrapper"
  >
    <div>
      <div>
        Layout
      </div>
      <div>
        <div>
          Next layout
        </div>
      </div>
    </div>
  </div>
</div>
`;
